pkg/network/vmicontroller/status.go

```diff
@@ -31,14 +31,15 @@ import (
 	"kubevirt.io/kubevirt/pkg/network/multus"
 	"kubevirt.io/kubevirt/pkg/network/namescheme"
 	"kubevirt.io/kubevirt/pkg/network/vmispec"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
 )
 
-func UpdateStatus(vmi *v1.VirtualMachineInstance, pod *k8scorev1.Pod) error {
+func UpdateStatus(clusterConfig *virtconfig.ClusterConfig, vmi *v1.VirtualMachineInstance, pod *k8scorev1.Pod) error {
 	var interfaceStatuses []v1.VirtualMachineInstanceNetworkInterface
 
 	networkStatuses := multus.NetworkStatusesFromPod(pod)
 
-	interfaceStatuses = append(interfaceStatuses, calculatePrimaryIfaceStatus(vmi, networkStatuses)...)
+	interfaceStatuses = append(interfaceStatuses, calculatePrimaryIfaceStatus(clusterConfig, vmi, networkStatuses)...)
```

Comment:
This is the beginning of the whole process, disable it and you've disabled everything.
```suggestion
if featureGateEnabled {
  interfaceStatuses = append(interfaceStatuses, calculatePrimaryIfaceStatus(clusterConfig, vmi, 
networkStatuses)...)
}
```

Reply:
If we do here the interface will not be populated, we need to skip the PrimaryPodInterfaceName field to be set .

Reply:
Nah  you right